Modules/mathmodule.c

```diff
@@ -1161,9 +1164,10 @@ FUNC2(copysign, copysign,
        "Return a float with the magnitude (absolute value) of x but the sign of y.\n\n"
       "On platforms that support signed zeros, copysign(1.0, -0.0)\n"
       "returns -1.0.\n")
-FUNC1(cos, cos, 0,
+FUNC1D(cos, cos, 0,
       "cos($module, x, /)\n--\n\n"
-      "Return the cosine of x (measured in radians).")
+      "Return the cosine of x (measured in radians).",
+      "expected a finite input, got %s")
```

Comment:
It works for NaN which is not a finite number.

Reply:
I don't think we should clutter the message with nans stuff.  It's expected, that math functions accept nans: garbage in - garbage out.

Reply:
"Finite" is not a colloquial term. It can be understood only in context of floating-point numbers, and there it means not infinity and not NaN.

Yes, other messages may need a correction too. Instead of "expected a number in range from -1 up to 1, got inf" it can say "inf is outside of range from -1 up to 1" or something like.

The error messages should be correct and unambiguous, otherwise there is no point in changing the old error messages.

Reply:
> But we _are_ in this context.

And in that context `math.isfinite()` returns false for NaN.

> Also, in such versions you will rather describe the _complement_ of the function domain. E.g. instead of "expected a nonnegative input, got -1.0" - something like "-1.0 is not a nonnegative number"

Rather "-1.0 is a negative integer number". If expression for the complement of the function domain is simpler, it should be used instead. In any case, all these messages should be reviewed by native speakers with mathematical or computer science background and teaching experience. cc @rhettinger, @tim-one, @mdickinson

Reply:
> Rather "-1.0 is a negative integer number".

With this wording it says something *about the input*, rather than about the function domain.

> all these messages should be reviewed by native speakers with mathematical or computer science background and teaching experience.

I can only appreciate that.

As summary, in the proposed patch we have error messages like "expected number in <some description of the function domain>, got %s", which is along the line, suggested by Raymond in the issue description.  Problem (well, maybe): this doesn't account the nan as an acceptable input.

Reply:
>  I don't think we should clutter the message with nans stuff. It's expected, that math functions accept nans: garbage in - garbage out.

I agree with @skirpichev: I don't think that we should explicitly mention NaN here.